modules/RTC/TrackStreamingStatus.js

```diff
@@ -0,0 +1,758 @@
+
+            // Listened which will be bound to JitsiRemoteTrack to listen for
+            // signalling mute/unmute events.
+            this._onSignallingMuteChanged
+                = this.onSignallingMuteChanged.bind(this);
+            this.track.on(
+                JitsiTrackEvents.TRACK_MUTE_CHANGED,
+                this._onSignallingMuteChanged);
+
+            // Used to send an analytics event when the video type changes.
+            this._onTrackVideoTypeChanged
+                = this.onTrackVideoTypeChanged.bind(this);
+            this.track.on(
+                JitsiTrackEvents.TRACK_VIDEOTYPE_CHANGED,
+                videoType => this._onTrackVideoTypeChanged(videoType));
+        }
+
+        this._onForwardedSourcesChanged = this._onForwardedSourcesChanged.bind(this);
+        this.conference.on(
+            JitsiConferenceEvents.FORWARDED_SOURCES_CHANGED,
+            this._onForwardedSourcesChanged);
+
+        this._onLastNValueChanged
+            = this.figureOutConnectionStatus.bind(this);
+        this.rtc.on(
+            RTCEvents.LASTN_VALUE_CHANGED, this._onLastNValueChanged);
+    }
+
+    /**
+     * Removes all event listeners and disposes of all resources held by this
+     * instance.
+     */
+    dispose() {
+        if (browser.supportsVideoMuteOnConnInterrupted()) {
+            this.rtc.removeListener(
+                RTCEvents.REMOTE_TRACK_MUTE,
+                this._onTrackRtcMuted);
+            this.rtc.removeListener(
+                RTCEvents.REMOTE_TRACK_UNMUTE,
+                this._onTrackRtcUnmuted);
+
+            this.track.off(
+                JitsiTrackEvents.TRACK_MUTE_CHANGED,
+                this._onSignallingMuteChanged);
+        }
+
+        this.conference.off(
+            JitsiConferenceEvents.FORWARDED_SOURCES_CHANGED,
+            this._onForwardedSourcesChanged);
+
+        this.rtc.removeListener(
+            RTCEvents.LASTN_VALUE_CHANGED, this._onLastNValueChanged);
+
+        this.conference.off(
+            JitsiConferenceEvents.P2P_STATUS, this._onP2PStatus);
+
+        this.conference.off(
+            JitsiConferenceEvents.USER_LEFT, this._onUserLeft);
+
+        this.clearTimeout();
+        this.clearRtcMutedTimestamp();
+        this.onUserLeft(this.track.getParticipantId());
+        this.figureOutConnectionStatus();
+    }
+
+    /**
+     * Changes connection status.
+     * @param newStatus
+     */
+    _changeConnectionStatus(newStatus) {
+        if (this.track.getConnectionStatus() !== newStatus) {
+
+            const sourceName = this.track.getSourceName();
+
+            this.track._setConnectionStatus(newStatus);
+
+            logger.debug(
+                `Emit track connection status(${Date.now()}) ${sourceName}: ${
+                    newStatus}`);
+
+            // Log the event on CallStats
+            Statistics.sendLog(
+                JSON.stringify({
+                    id: 'track.conn.status',
```

Comment:
Is there another part in the stack that needs to be updated in order for CallStats to handle this new id?

Reply:
I don't think this part is particularly important (it only logs a custom event on CS timeline), but lets change the `id` to be `track.streaming.status`